src/ensembl/src/content/app/browser/track-panel/track-panel-bar/TrackPanelBarIcon.tsx

```diff
@@ -20,51 +14,49 @@ type TrackPanelBarIconProps = {
   trackPanelModalView: string;
 };
 
-const TrackPanelBarIcon: FunctionComponent<TrackPanelBarIconProps> = memo(
-  (props: TrackPanelBarIconProps) => {
-    const [toggleState, setToggleState] = useState(false);
-
-    useEffect(() => {
-      if (!props.isTrackPanelModalOpened) {
-        setToggleState(false);
-      }
-    }, [props.isTrackPanelModalOpened]);
-
-    const toggleModalView = useCallback(() => {
-      if (!props.isTrackPanelOpened) {
-        return;
-      }
-
-      const newToggleState = !toggleState;
-
-      if (newToggleState) {
-        props.openTrackPanelModal(props.iconConfig.name);
-      } else {
-        props.closeTrackPanelModal();
-      }
-
-      setToggleState(newToggleState);
-    }, [props.iconConfig.name, props.isTrackPanelOpened, toggleState]);
-
-    const getViewIconStatus = () => {
-      const { iconConfig, trackPanelModalView } = props;
-
-      return iconConfig.name === trackPanelModalView
-        ? ImageButtonStatus.HIGHLIGHTED
-        : ImageButtonStatus.ACTIVE;
-    };
-
-    return (
-      <dt className={styles.barIcon}>
-        <ImageButton
-          buttonStatus={getViewIconStatus()}
-          description={props.iconConfig.description}
-          onClick={toggleModalView}
-          image={props.iconConfig.icon}
-        />
-      </dt>
-    );
-  }
-);
+const TrackPanelBarIcon = memo((props: TrackPanelBarIconProps) => {
```

Comment:
`memo` is a premature optimization.

Reply:
Ah the old code has come back to haunt us 😄 I'll remove these.